spec/support/unit/record_with_unrelated_error_builder.rb

```diff
@@ -0,0 +1,90 @@
+require_relative 'helpers/model_builder'
+
+module UnitTests
+  class RecordWithUnrelatedErrorBuilder
+    include ModelBuilder
+
+    def initialize(options)
+      @options = options.reverse_merge(default_options)
+    end
+
+    def attribute_that_receives_error
+      options[:attribute_that_receives_error]
+    end
+
+    def attribute_to_validate
+      options[:attribute_to_validate]
+    end
+
+    def message
+      options[:message]
+    end
+
+    def message=(message)
+      options[:message] = message
+    end
+
+    def model
+      @_model ||= create_model
+    end
+
+    def model_name
+      'Example'
+    end
+
+    def record
+      model.new
+    end
+
+    def valid_value
+      'some value'
+    end
+
+    protected
+
+    attr_reader :options
+
+    private
+
+    def context
+      {
+        validation_method_name: validation_method_name,
+        valid_value: valid_value,
+        attribute_to_validate: attribute_to_validate,
+        attribute_that_receives_error: attribute_that_receives_error,
+        message: message
+      }
+    end
+
+    def create_model
+      _context = context
+
+      define_model model_name, model_columns do
+        validate _context[:validation_method_name]
+
+        define_method(_context[:validation_method_name]) do
+          self.errors.add(_context[:attribute_that_receives_error], _context[:message])
```

Comment:
This is similar to RecordWithDifferentErrorAttributeBuilder, but the behavior I want is for there to be an error that has nothing to do with the attribute to validate. If there's a better way to shoehorn that in the other class *or* to represent this condition in a builder, please let me know.

Reply:
I think this is fine!